pkgs/development/misc/resholve/deps.nix

```diff
@@ -0,0 +1,150 @@
+  a libc test oil runs. My oil fork just disabled the libc tests, but
+  I haven't quite decided if that's the right long-term call, so I
+  didn't add a patch for it here yet.
+*/
+
+rec {
+  # had to add this as well; 1.3 causes a break here; sticking
+  # to oil's official 1.0.3 dep for now.
+  re2c = stdenv.mkDerivation rec {
+    pname = "re2c";
+    version = "1.0.3";
+    sourceRoot = "${src.name}/re2c";
+    src = fetchFromGitHub {
+      owner = "skvadrik";
+      repo = "re2c";
+      rev = version;
+      sha256 = "0grx7nl9fwcn880v5ssjljhcb9c5p2a6xpwil7zxpmv0rwnr3yqi";
+    };
+    nativeBuildInputs = [ autoreconfHook ];
+    preCheck = ''
+      patchShebangs run_tests.sh
+    '';
+  };
+
+  py-yajl = python27.pkgs.buildPythonPackage rec {
+    pname = "oil-pyyajl";
+    version = "unreleased";
+    src = fetchFromGitHub {
+      owner = "oilshell";
+      repo = "py-yajl";
+      rev = "eb561e9aea6e88095d66abcc3990f2ee1f5339df";
+      sha256 = "17hcgb7r7cy8r1pwbdh8di0nvykdswlqj73c85k6z8m0filj3hbh";
+      fetchSubmodules = true;
+    };
+    # just for submodule IIRC
+    nativeBuildInputs = [ git ];
+  };
+
+  # resholve's primary dependency is this developer build of the oil shell.
+  oildev = python27.pkgs.buildPythonPackage rec {
+    pname = "oil";
+    version = "undefined";
+
+    src = fetchFromGitHub {
+      owner = "oilshell";
+      repo = "oil";
+      rev = "ea80cdad7ae1152a25bd2a30b87fe3c2ad32394a";
+      sha256 = "0pxn0f8qbdman4gppx93zwml7s5byqfw560n079v68qjgzh2brq2";
+
+      /*
+      This feels a little dumb; I assume it's wrongish. I was hoping
+      to be able to filter the source down to knock out parts of the
+      source that I'm not using, but I get errors just trying to get
+      *anything* working with the approaches commented out below.
+
+      On IRC eadwu[m] suggested dropping them in extraPostFetch.
+
+      It's not critical to drop most of these; the primary target is
+      the vendored fork of Python-2.7.13, which is ~ 55M and over 3200
+      files, dozens of which get interpreter script patches in fixup.
+      */
+      extraPostFetch = ''
+        #find . -maxdepth 1 -type d | sort
+        rm -rf Python-2.7.13 benchmarks metrics py-yajl rfc gold web testdata services demo devtools cpp
+        # find . -maxdepth 1 -type d | sort
+      ''; # breakers: doc, pgen2, vendor
+    };
+
+    # src = stdenv.lib.cleanSourceWith {
+    #   src = fetchFromGitHub {
+    #     owner = "oilshell";
+    #     repo = "oil";
+    #     rev = "ea80cdad7ae1152a25bd2a30b87fe3c2ad32394a";
+    #     sha256 = "0pxn0f8qbdman4gppx93zwml7s5byqfw560n079v68qjgzh2brq2";
+    #   };
+    #   filter = stdenv.lib.cleanSourceFilter;
+    # };
+
+    # src = stdenv.lib.sourceByRegex (fetchFromGitHub {
+    #   owner = "oilshell";
+    #   repo = "oil";
+    #   rev = "ea80cdad7ae1152a25bd2a30b87fe3c2ad32394a";
+    #   sha256 = "0pxn0f8qbdman4gppx93zwml7s5byqfw560n079v68qjgzh2brq2";
+    # }) [".*"];
```

Comment:
Why all this commented code?

Reply:
```
/*
      This feels a little dumb; I assume it's wrongish. I was hoping
      to be able to filter the source down to knock out parts of the
      source that I'm not using, but I get errors just trying to get
      *anything* working with the approaches commented out below.

      On IRC eadwu[m] suggested dropping them in extraPostFetch.

      It's not critical to drop most of these; the primary target is
      the vendored fork of Python-2.7.13, which is ~ 55M and over 3200
      files, dozens of which get interpreter script patches in fixup.
      */
```

Reply:
Well, I understand the issue, but I still won't sure that having commented code really helps.

Maybe adding just a note that you tried to use `stdenv.lib.cleanSourceWith` and `stdenv.lib.sourceByRegex` in this bigger comment and it didn't work?

Reply:
The comments can/should/will be removed before merge if someone says the way I'm doing it is fine (or points out how to do it correctly).

Reply:
Ok, fair enough. It is just that I don't know if your PR is ready to review and merge or not.

It seems like you're looking more at feedback about it and less about trying to merge as is, did I understand it right?

Reply:
Because if it is, probably my review isn't help here.

Reply:
Sorry--I believe it is roughly ready, but it is my first full package and it's covering some complexish ground where I had no clue if my decisions were appropriate. 

You are someone! If you say filtering them with `extraPostFetch` is fine by you, I'll cut the comments. If you aren't sure, I'll find someone to help clarify it.

Reply:
For me it is fine, don't see the issue really.

Reply:
BTW, I really would like if you could remove other comments from your code this would help. I think those comments makes the review harder.

You could for example convert those commented codes in inline comments like we are doing here.

Reply:
I pared comments back in 240bf06--is that what you had in mind?